lang/java/tools/src/main/java/org/apache/avro/tool/SpecificCompilerTool.java

```diff
@@ -90,13 +107,13 @@ public int run(InputStream in, PrintStream out, PrintStream err,
       Schema.Parser parser = new Schema.Parser();
       for (File src : determineInputs(inputs, SCHEMA_FILTER)) {
         Schema schema = parser.parse(src);
-        SpecificCompiler compiler = new SpecificCompiler(schema);
+        SpecificCompiler compiler = new SpecificCompiler(schema, dateTimeLogicalTypeType.orElse(DateTimeLogicalTypeType.JODA));
```

Comment:
is there a reason why a constructor parameter is used here? configuring decimal support seems a similar use case which is handled by a setter parameter. it would be more consistent to use a setter parameter here, and probably result in a smaller change.

Reply:
Apologies - just saw your reasoning in your summary. Please ignore.